src/retrieval/utils.rs

```diff
@@ -0,0 +1,15 @@
+use hdk::prelude::*;
+
+/// A triple of an Entry along with the HeaderHash
+/// of that committed entry and the EntryHash of the entry
+pub type EntryAndHash<T> = (T, HeaderHash, EntryHash);
+
+/// The same as an EntryAndHash but inside an Option,
+/// so it can be Some(...) or None
+pub type OptionEntryAndHash<T> = Option<EntryAndHash<T>>;
```

Comment:
would prefer these to be in a spot like 'types' or something, but not going to ask that to be done right now.